plugins/modules/foreman_role.py

```diff
@@ -101,7 +118,21 @@ def main():
         if 'organizations' in entity_dict:
             entity_dict['organizations'] = module.find_resources_by_name('organizations', entity_dict['organizations'], thin=True)
 
-    changed = module.ensure_entity_state('roles', entity_dict, entity)
+    filters = entity_dict.pop("filters", None)
+
+    changed, entity = module.ensure_entity('roles', entity_dict, entity)
+
+    if not module.desired_absent:
+        if filters is not None:
+            existing_filters = entity['filters']
+            for filter in filters:
```

Comment:
I think `filter` is a built in function in Python, so we might want to use some other name here

Reply:
replace with `f` :)

Reply:
I would've went with `role_filter` ;)

Reply:
Right, that's much better. Golang infected my mind. Updated.